lib/index.ts

```diff
@@ -662,15 +648,19 @@ export default class Backburner {
 
   private _scheduleExpiredTimers() {
     let timers = this._timers;
-    let l = timers.length;
     let i = 0;
+    let l = timers.length;
     let defaultQueue = this.options.defaultQueue;
     let n = this._platform.now();
-    for (; i < l; i += 2) {
+
+    for (; i < l; i += 5) {
       let executeAt = timers[i];
       if (executeAt <= n) {
-        let fn = timers[i + 1];
-        this.schedule(defaultQueue, null, fn);
+        let target = timers[i + 2];
+        let method = timers[i + 3];
+        let args = timers[i + 4];
+        let stack = this.DEBUG ? new Error() : undefined;
```

Comment:
This is being done in the wrong place. `_scheduleExpiredTimers` is _after_ the real `setTimeout` has ran and we are scheduling the jobs into the actions queue.  That will mean the debug stack we are grabbing here is now the stack from where `run.later(...)` was called.

This needs to be moved into `_setTimeout` where we are doing `this._timers.splice(...)` / `this._timers.push` so that the captured debug stack is the "right" place.

Reply:
OK, pushed a fix for this issue to the branch along with a test...